ros/src/tl_detector/tl_detector.py

```diff
@@ -269,18 +269,63 @@ def get_light_state(self, light):
         # Get classification
         return self.light_classifier.get_classification(img_traffic_light)
 
-    def get_tl_waypoints(self):
+    def get_tl_waypoints_idx(self):
         """ Converts array self.lights with trafic
         light positions to get_closest_waypoint_idx array with PoseStamped
         traffic light waypoints
-
         """
 
         for light in self.lights:
-            self.tl_waypoints.append(
+            self.tl_waypoints_idx.append(
                 self.get_closest_waypoint_idx(light.pose.pose)
             )
 
+    def get_closest_tl_wp_idx(self, car_pos_wp_idx, searching_dist_tl):
+        """ Finds closest traffic light waypoint in searching range
+
+        Args:
+            car_pos_wp_idx (Integer): current position of the car
+            searching_dist_tl: number of waypoints that will be searched
+            for a traffic light
+
+        Returns:
+            light_wp_idx: waypoint index of closest traffic light in range
+
+        """
+
+        light_wp_idx = -1
+        # In case that array with traffic light waypoints does not exist,
+        # create it
+        if(self.tl_waypoints_idx == []):
+            self.get_tl_waypoints_idx()
+        # Check distance between car and traffic light to address end of array
+        # issue
+        last_waypoint_index = len(self.waypoints.waypoints) - 1
+        car_distance_from_last_wp = last_waypoint_index - car_pos_wp_idx
+        # Loop thorugh waypoints to find the closest traffic ligt waypoint
+        smallest_tl_distance = 10000
+        for tl_waypoint_idx in self.tl_waypoints_idx:
+            # Cover corner case where traffic light is just behind the
+            # waypoint 0 And the car is near last waypoint so waypoint
+            # index will reset to 0
+            if(searching_dist_tl > (car_distance_from_last_wp) and
```

Comment:
Here the 80 characters line limit makes the code hard to read... Could we bump up the limit to 120 characters? @carlosgalvezp